libs/sdk-core/src/greenlight/node_api.rs

```diff
@@ -684,6 +684,65 @@ impl Greenlight {
         _ = self.inprogress_payments.fetch_sub(1, Ordering::Relaxed);
         res
     }
+
+    async fn wait_channel_reestablished(&self, path: &PaymentPath) -> NodeResult<()> {
+        let deadline = match Instant::now().checked_add(Duration::from_secs(20)) {
+            Some(deadline) => deadline,
+            None => return Err(NodeError::generic("failed to add 20 seconds to now")),
```

Comment:
Bit more informative
```suggestion
            None => return Err(NodeError::generic("Failed to set channel establishment deadline")),
```